include/alpaka/exec/ExecCpuOmp2Threads.hpp

```diff
@@ -169,27 +169,27 @@ namespace alpaka
                         // Therefore we use 'omp parallel' with the specified number of threads in a block.
                         #pragma omp parallel num_threads(iBlockThreadCount)
                         {
-#if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
-                            // GCC 5.1 fails with:
-                            // error: redeclaration of const int& iBlockThreadCount
-                            // if(numThreads != iBlockThreadCount
-                            //                ^
-                            // note: const int& iBlockThreadCount previously declared here
-                            // #pragma omp parallel num_threads(iBlockThreadCount)
-                            //         ^
-#if (!BOOST_COMP_GNUC) || (BOOST_COMP_GNUC < BOOST_VERSION_NUMBER(5, 0, 0))
                             // The first thread does some checks in the first block executed.
                             if((::omp_get_thread_num() == 0) && (acc.m_gridBlockIdx.sum() == 0u))
```

Comment:
You will have to change the `acc.m_gridBlockIdx.sum() == 0u` to `acc.m_gridBlockIdx.sum() == static_cast<TIdx>(0)` to makethe compiler happy.